src/main/database.cpp

```diff
@@ -68,6 +84,21 @@ void Database::setLoggingLevel(std::string loggingLevel) {
     spdlog::set_level(LoggingLevelUtils::convertStrToLevelEnum(std::move(loggingLevel)));
 }
 
+void Database::checkAccessMode() {
+    if (!FileUtils::fileOrPathExists(databasePath)) {
+        FileUtils::createDir(databasePath);
+    }
+    int flags;
+    FileLockType lock;
+    auto lockFilePath = StorageUtils::getLockFilePath(databasePath);
+    if (!FileUtils::fileOrPathExists(lockFilePath)) {
+        getFileFlags(systemConfig, flags, lock, true);
```

Comment:
1. You're setting both `flags` and `lock`, why name this function as `getFileFlags`?